### DISCUSSION TITLE
Modularizing the Quarkus-CXF project

### DISCUSSION THREAD

**Discussion by shumonsharif** [quarkiverse/quarkus-cxf#375] [Ideas]
@dufoli @famod This will entail a fairly large refactoring of the code base - so was hoping for some discussion on the topic.

Basically, the proposal here is to modularize the existing code and split it up into sub modules that align with the corresponding CXF modules.  

This should result in a significantly cleaner and well-structured codebase, while allowing for growth as we add support for additional CXF modules.  An example list of modules we can start off with for quarkus-cxf is provided below, please do share your thoughts?

- quarkus-cxf
- quarkus-cxf-rt-features-logging
- quarkus-cxf-rt-features-metrics
- quarkus-cxf-rt-ws-addr
- quarkus-cxf-rt-ws-policy
- quarkus-cxf-rt-ws-transfer

**Comment by shumonsharif** [comment]
Also, cc: @gastaldi - Are there any good Quarkiverse projects that follow a similar structure (ie. many sub-modules) within a single project?  Wondering whether we can pull any ideas in from others, especially around how to handle documentation, automated builds, etc.

**Reply by gastaldi** [reply]
Yes, the following repositories are good examples:

- https://github.com/quarkiverse/quarkus-amazon-services
- https://github.com/quarkiverse/quarkus-micrometer-registry
- https://github.com/quarkiverse/quarkus-config-extensions
- https://github.com/quarkiverse/quarkus-google-cloud-services

**Reply by shumonsharif** [reply]
Thanks for all the links; will go through these!
